### PR TITLE
Use IUnconfiguredProjectVsServices instead of HostObject

### PR DESCRIPTION
**Pull request by davkean**
- Use IUnconfiguredProjectVsServices instead of HostObject
- Also inlined initialization and removed an unneeded Requires (it would null ref before it was called).

### CODE REVIEW

**Comment by davkean** [comment]
tag @dotnet/project-system @mavasani

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/UnconfiguredProjectHostObject.cs [hunk 1]
```diff
@@ -11,16 +11,12 @@ namespace Microsoft.VisualStudio.ProjectSystem.VS.LanguageServices
...
     {
-        private readonly Dictionary<uint, IVsHierarchyEvents> _hierEventSinks;
-        private readonly HashSet<uint> _pendingItemIds;
+        private readonly Dictionary<uint, IVsHierarchyEvents> _hierEventSinks = new Dictionary<uint, IVsHierarchyEvents>();
+        private readonly HashSet<uint> _pendingItemIds = new HashSet<uint>();
 
-        public UnconfiguredProjectHostObject(UnconfiguredProject unconfiguredProject)
-            : base (innerHierarchy: (IVsHierarchy)unconfiguredProject.Services.HostObject, innerVsProject: (IVsProject)unconfiguredProject.Services.HostObject)
+        public UnconfiguredProjectHostObject(IUnconfiguredProjectVsServices projectVsServices)
+            : base(innerHierarchy: projectVsServices.VsHierarchy, innerVsProject: projectVsServices.VsProject)
```
Comment (srivatsn) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/UnconfiguredProjectHostObject.cs]: For my education - what does this fix?

Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/UnconfiguredProjectHostObject.cs]: It doesn't "fix" anything, the behavior is exactly the same but reduces the coupling that "HostObject" implements the IVsHierarchy and IVsProject. Only IUnconfiguredProjectVsServices knows that. We'll be thankful if/when we bring these components up in other IDEs.

Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/UnconfiguredProjectHostObject.cs]: Was walking through this code - and noticed it, so fixed it while I was there.

Comment (srivatsn) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/UnconfiguredProjectHostObject.cs]: Makes sense.

Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/UnconfiguredProjectHostObject.cs]: Yep, looks like a good sign that I should add context in the commit for the future.

**Pull request merged**